crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4142,23 +4159,46 @@ mod tests {
         assert_public_ty(&db, "src/a.py", "d", "Literal[-1]");
         assert_public_ty(&db, "src/a.py", "e", "Literal[2]");
 
+        // More division cases
         db.write_dedented(
             "/src/b.py",
             "
             a = 1 / 2
             b = 1 // 2
             c = 4 / 2
             d = 4 // 2
-            e = 4 // 0
             ",
         )?;
 
         assert_public_ty(&db, "/src/b.py", "a", "float");
         assert_public_ty(&db, "/src/b.py", "b", "Literal[0]");
         assert_public_ty(&db, "/src/b.py", "c", "float");
         assert_public_ty(&db, "/src/b.py", "d", "Literal[2]");
-        // TODO: division by zero should be an error
-        assert_public_ty(&db, "/src/b.py", "e", "int");
+
+        // Division by zero
```

Comment:
I would rather have this as a separate test case function.

Though since it will all change with test framework anyway, it's OK to not bother.